CMake/RealmCore.cmake

```diff
@@ -179,7 +180,11 @@ function(download_realm_core core_version)
         download_realm_tarball(${url_release} ${core_directory_release} ${core_library_release})
     else()
         if(APPLE)
-            set(basename "realm-core-cocoa")
+            if(REALM_CORE_PACKAGING STREQUAL "1")
```

Comment:
You could probably use `if(${core_version} VERSION_GREATER_EQUAL "3.0.0")` instead of adding the extra flag.